sys/fs/cuse/cuse.c

```diff
@@ -1133,7 +1149,18 @@ cuse_server_ioctl(struct cdev *dev, unsigned long cmd,
 
 		pccmd->entered = curthread;
 
-		*pcmd = pccmd->sub;
+#ifdef COMPAT_FREEBSD64
+		if (!SV_CURPROC_FLAG(SV_CHERI)) {
```

Comment:
Move this up to between lines 1129 and 1130, then you don't need to check the ABI, you can just use the ioctl to determine it. Checking the ABI should only be done when the ioctls have the same encoding for both.